### PR TITLE
Adds a roomba variant

### PR DESCRIPTION
**Pull request by ghost**
Created by using a maid headband on a roomba. Can be armed with a kitchen knife instead of a lighter/fork, and does slightly higher melee damage.

All credit for the sprites goes to basil.

dreammaker_VegQTpaUYf

### CODE REVIEW

File: code/game/machinery/bots/cleanbot.dm [hunk 1]
```diff
@@ -392,3 +397,33 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
...
+	armed = 0
+
+/obj/machinery/bot/cleanbot/roomba/meido/attackby(var/obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/kitchen/utensil/fork) || istype(W, /obj/item/weapon/lighter))
+		to_chat(user, "<span class='notice'>\the [src] buzzes and recoils at \the [W]. Perhaps it would prefer something more refined?</span>")
+		return
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		to_chat(user, "<span class='notice'>\the [src] is already wearing one of those!</span>")
+		return
+	else if(istype(W, /obj/item/weapon/kitchen/utensil/knife/large) && !armed && user.a_intent != I_HURT)
```
Comment (boy2mantwicethefam) [inline code/game/machinery/bots/cleanbot.dm]: Large knife variants include the thematically-cultish dagger, the kitchen cleaver and the antagonist cleaver

Comment (ghost) [reply code/game/machinery/bots/cleanbot.dm]: destroy all oldcoders

File: code/game/machinery/bots/cleanbot.dm [hunk 2]
```diff
@@ -271,6 +271,11 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
+		to_chat(user, "<span class='notice'>You attach \the [W] to \the [src]. It makes a soft booping noise and its light blinks excitedly for a moment.</span>")
+		qdel(src)
```
Comment (beesil) [inline code/game/machinery/bots/cleanbot.dm:284]: If there's a pAI in the roomba, will this delete it?

Comment (ghost) [reply code/game/machinery/bots/cleanbot.dm:284]: Yes, nice catch.

**Comment by PrimeDSS13** [comment]
It's missing the pAI blue cross, I believe. Very nice though.

**Comment by ghost** [comment]
Apparently only the medibot and mulebot have overlays for pAIs, that could be addressed in a separate PR.

**Comment by DeityLink** [comment]
make it stop time too

File: code/game/machinery/bots/cleanbot.dm [hunk 3]
```diff
@@ -271,6 +271,15 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
+		if(integratedpai)
```
Comment (DamianX) [inline code/game/machinery/bots/cleanbot.dm]:
```suggestion
		M.install_pai(eject_integrated_pai_if_present())
```
This should also correctly transfer the verb. I suggesting testing this

Comment (ghost) [reply code/game/machinery/bots/cleanbot.dm]: Seems to work fine

**Comment by tuypo1** [comment]
It's just like one of my japanese animes

File: code/game/machinery/bots/cleanbot.dm [hunk 4]
```diff
@@ -392,3 +398,33 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
...
+	icon_state = "maidbot0"
+	icon_initial = "maidbot"
+	armed = 0
+
+/obj/machinery/bot/cleanbot/roomba/meido/attackby(var/obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/kitchen/utensil/fork) || istype(W, /obj/item/weapon/lighter))
+		to_chat(user, "<span class='notice'>\the [src] buzzes and recoils at \the [W]. Perhaps it would prefer something more refined?</span>")
+		return
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		to_chat(user, "<span class='notice'>\the [src] is already wearing one of those!</span>")
```
Comment (Zth--) [inline code/game/machinery/bots/cleanbot.dm]: I think it should be
```suggestion
		to_chat(user, "<span class='notice'>\The [src] is already wearing one of those!</span>")
```
same for line 411 and 419

File: code/game/machinery/bots/cleanbot.dm [hunk 5]
```diff
@@ -271,6 +271,12 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
```
Comment (DamianX) [inline code/game/machinery/bots/cleanbot.dm]: won't this result in all unnamed maidbots being named cleanbot? It's a very minor thing but you could check if the name is equal to the initial name first.

Comment (ghost) [reply code/game/machinery/bots/cleanbot.dm]: tru

**Pull request merged**